app/upgrades/v17/upgrades.go

```diff
@@ -130,6 +130,50 @@ func CreateUpgradeHandler(
 			CosmwasmWeight:     300,
 		})
 
+		// get all the existing CL pool Ids
+		poolIds, err := keepers.ConcentratedLiquidityKeeper.GetPoolIds(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		// migrate twap records for CL Pools
+		err = FlipTwapSpotPriceRecords(ctx, poolIds, keepers)
+		if err != nil {
+			return nil, err
+		}
+
```

Comment:
This will not work for the following reason:

in the v17 binary, the spot price will be properly flipped. This means that any pools created in the upgrade handler will be correct. Above this code we create a bunch of CL pools. These CL pools will be apart of this record flip.

It likely wont be a big deal because there will probably be only one record at time of creation but, we are unnecessarily changing all these new CL pool's TWAPs which dont need to be flipped

Reply:
this makes sense, will move this logic at the top of migration